packages/event-handler/src/appsync-graphql/Router.ts

```diff
@@ -0,0 +1,229 @@
+import {
+  EnvironmentVariablesService,
+  isRecord,
+} from '@aws-lambda-powertools/commons';
+import type {
+  GenericLogger,
+  GraphQlRouteOptions,
+  GraphQlRouterOptions,
+  OnMutationHandler,
+  OnQueryHandler,
+} from '../types/appsync-graphql.js';
+import { RouteHandlerRegistry } from './RouteHandlerRegistry.js';
+
+/**
+ * Class for registering routes for the `query` and `mutation` events in AWS AppSync GraphQL APIs.
+ */
+class Router {
+  /**
+   * A map of registered routes for the `query` event, keyed by their fieldNames.
+   */
+  protected readonly onQueryRegistry: RouteHandlerRegistry;
+  /**
+   * A map of registered routes for the `mutation` event, keyed by their fieldNames.
+   */
+  protected readonly onMutationRegistry: RouteHandlerRegistry;
+  /**
+   * A logger instance to be used for logging debug, warning, and error messages.
+   *
+   * When no logger is provided, we'll only log warnings and errors using the global `console` object.
+   */
+  protected readonly logger: Pick<GenericLogger, 'debug' | 'warn' | 'error'>;
+  /**
+   * Whether the router is running in development mode.
+   */
+  protected readonly isDev: boolean = false;
+  /**
+   * The environment variables service instance.
+   */
+  protected readonly envService: EnvironmentVariablesService;
+
+  public constructor(options?: GraphQlRouterOptions) {
+    this.envService = new EnvironmentVariablesService();
+    const alcLogLevel = this.envService.get('AWS_LAMBDA_LOG_LEVEL');
```

Comment:
We are in the process of simplifying the way we handle environment variables, if possible let's switch to [this pattern](https://github.com/aws-powertools/powertools-lambda-typescript/blob/351e88e34fb5650239a97c400e3a339c2c60f2a1/packages/event-handler/src/bedrock-agent/BedrockAgentFunctionResolver.ts#L54-L57) instead (you might have to rebase/rebuild the commons package for it to be available).